ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue

```diff
@@ -55,39 +73,51 @@ export default defineComponent({
   props: {
     groupId: { type: String },
     fractionalData: { type: Object },
-    isAddingHomeOwner: { type: Boolean } // make additional Group available in dropdown when adding a new home owner
+    isAddingHomeOwner: { type: Boolean } // makes additional Group available in dropdown when adding a new home owner
   },
+
   setup (props, { emit }) {
-    const { getMhrRegistrationHomeOwnerGroups } = useGetters<any>([
-      'getMhrRegistrationHomeOwnerGroups'
-    ])
+    const { getMhrRegistrationHomeOwnerGroups } = useGetters<any>(['getMhrRegistrationHomeOwnerGroups'])
     const groupDropdown = ref(null)
 
     const { required } = useInputRules()
     const { getGroupDropdownItems, showGroups } = useHomeOwners()
+    const homeOwnerGroups = [...getMhrRegistrationHomeOwnerGroups.value]
 
     const localState = reactive({
       ownerGroupId: props.groupId,
       groupItems: computed(() => getGroupDropdownItems(props.isAddingHomeOwner, props.groupId)),
       groupRules: computed(() => {
         return showGroups.value ? required('Select a group for this owner') : []
       }),
+      groupFractionalData: find(getMhrRegistrationHomeOwnerGroups.value, { groupId: props.groupId }),
+      fractionalInfo: computed(() => props.fractionalData),
       showFractionalOwnership: computed(() => Number(localState.ownerGroupId) > 0),
-      isReadOnlyFractionalOwnership: computed(
-        // if group already exists - show fractional ownership as readonly
-        () =>
-          find(
-            getMhrRegistrationHomeOwnerGroups.value,
-            group => group.groupId === localState.ownerGroupId
-          ) !== undefined
-      )
+      allGroupsState: getMhrRegistrationHomeOwnerGroups.value
```

Comment:
Type might be helpful here as well, since the end result appears to have some complexity.

Reply:
Added interface.